controllers/taloscontrolplane_controller.go

```diff
@@ -386,13 +389,19 @@ func (r *TalosControlPlaneReconciler) bootControlPlane(ctx context.Context, clus
 }
 
 func (r *TalosControlPlaneReconciler) generateTalosConfig(ctx context.Context, tcp *controlplanev1.TalosControlPlane, cluster *capiv1.Cluster, spec *cabptv1.TalosConfigSpec) (*corev1.ObjectReference, error) {
+	owner := metav1.OwnerReference{
+		APIVersion:         controlplanev1.GroupVersion.String(),
+		Kind:               "TalosControlPlane",
+		Name:               tcp.Name,
+		UID:                tcp.UID,
+		BlockOwnerDeletion: to.BoolPtr(true),
```

Comment:
you probably might want to use https://godoc.org/k8s.io/utils/pointer so that you don't have pull Azure